### DISCUSSION TITLE
[FEATURE] Better interface for result_callback

### DISCUSSION THREAD

**Discussion by tupui** [fastapi/typer#881] [Questions]
Thanks for the amazing library! I have a suggestion about result callbacks.

Currently, if you want to have a callback after a command, you can use the `result_callback` argument of `@app.callback`. This is not described in the doc and a bit of a hidden feature. Consider the following:

```python
import typer

app = typer.Typer()


def tear_down(*args, **kwargs):
    print("after")


@app.callback(result_callback=tear_down)
def setup():
    print("before")

@app.command()
def my_command():
    print("command call")

# before
# command call
# after
```

It works but the interface could look clearer. I am confused that the result callback has to be defined with the callback.
I would prefer to have either a dedicated `@app.resultcallback` (as click does). Or a parameter on callback `@app.callback(result_callback=True)`. But I would definitely dissociate both callback definitions from each other.

Also, IMO this should be documented in the callback section of the doc. I was looking into this for a proper way to setup and close DB connections.

**Comment by YuriiMotov** [comment]
You can also specify it as a `result_callback` parameter of `typer.Typer()`

```py
app = typer.Typer(result_callback=tear_down)
```
